### PR TITLE
RHDEVDOCS-4259 GCP CLF for logging 5.5.1

### PR DESCRIPTION
**Pull request by libander**
Version(s): 4.10+

Issue: https://issues.redhat.com/browse/RHDEVDOCS-4259

Link to docs preview: https://drive.google.com/file/d/1rZByPAvvYxnwdNZryCVoZ7bHRB3zkF5p/view?usp=sharing

Reviewers: 
SME:@alanconway 
QE: @kabirbhartiRH / @QiaolingTang / @anpingli

### CODE REVIEW

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  namespace: "openshift-logging"
+spec:
+  outputs:
+    - name: gcp-1
+      type: googleCloudLogging
+      secret:
+        name: gcp-secret
+      googleCloudLogging: <1>
+        projectId : "openshift-gce-devel"
+        logId : "app-gcp"
```
Comment (alanconway) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:30]:
```suggestion
        logId : "app-gcp" <2>
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - name: test-app
+      inputRefs:
+        - application
+          #        - infrastructure
+      outputRefs:
+        - gcp-1
+----
+<1> billingAccountID, folderID, organizationID, or projectID must be set //differences? why pick one over another?
+<3>
+<4>
```
Comment (alanconway) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:41]:
```suggestion
<2> The logId is added to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry]
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc (hunk 1 above)
Comment (alanconway) [reply modules/cluster-logging-collector-log-forward-gcp.adoc:30]: logId, projectId etc. shoud be "logID", "projectID" - if we've released the code that way already I guess we have to live with it.

**Review by alanconway — CHANGES_REQUESTED** [review]
Minor changes.

File: modules/cluster-logging-collector-log-forward-gcp.adoc (hunk 1 above)
Comment (libander) [reply modules/cluster-logging-collector-log-forward-gcp.adoc:30]: https://github.com/openshift/cluster-logging-operator/pull/1614/files#diff-8f9105693decf749520b1f29a67276eef07601e8873521c61d38a5b93e667ec6

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    //LogID is the log ID to which to publish logs. This identifies log stream.
+    //+optional
+    LogID string `json:"logId,omitempty"`
+
+    SeverityKey string `json:"severityKey,omitempty"`
+}
+----
+
+
```
[the remark this answers is not in the document]
Comment (libander) [reply modules/cluster-logging-collector-log-forward-gcp.adoc:61]: We don't, this was still marked draft.

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
+:_content-type: PROCEDURE
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:1]:
```suggestion
// Module included in the following assemblies:
//
// * 
:_content-type: PROCEDURE
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+[id="cluster-logging-collector-log-forward-gcp_{context}"]
+= Forwarding logs to Google Cloud Platform
+
+You can forward logs to link:https://cloud.google.com/logging/docs/basic-concepts[Google Cloud Logging] in addition to, or instead of, the internal default {product-title} log store.
+
+[NOTE]
+====
+Using this feature with fluentd is not supported.
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:9]:
```suggestion
Using this feature with Fluentd is not supported.
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+====
+
+.Prerequisites
+* {logging-title-uc} Operator 5.5.1 and later
+
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:14]:
```suggestion

.Procedure

. <first step> etc.
+
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        logId : "app-gcp" <2>
+  pipelines:
+    - name: test-app
+      inputRefs:
+        - application
+          #- infrastructure
+      outputRefs:
+        - gcp-1
+----
+<1> Exactly one of projectId, folderId, organizationId, or billingAccountId must be set depending on where you want to store your logs in the link:https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy[GCP resource hierarchy].
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:39]:
```suggestion
<1> Set either a `projectId`, `folderId`, `organizationId`, or `billingAccountId` stanza, depending on where you want to store your logs in the link:https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy[GCP resource hierarchy].
```
- Used active voice, put backticks around items that appear in code, and made them adjectives of the noun, "stanza."

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+<2> The logId is added to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry].
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:40]:
```suggestion
<2> Set the `logId` value to add to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry].
```
Made this parallel to the first callout.

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+== Additional Resources
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:42]:
```suggestion
[role="_additional-resources"]
.Additional Resources
```
Modules have an `.Additional resources` heading with a `role` attribute. Assemblies have an `== Additional resources` heading with a unique ID.

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 10]
```diff
@@ -0,0 +1,44 @@
+:_content-type: PROCEDURE
+[id="cluster-logging-collector-log-forward-gcp_{context}"]
+= Forwarding logs to Google Cloud Platform
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:3]:
```suggestion
= Forwarding logs to Google Cloud Platform (GCP)
```

**Comment by anpingli** [comment]
What does the GCP secret  look like? it is better to provide an example.

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+kind: "ClusterLogForwarder"
+metadata:
+  name: "instance"
+  namespace: "openshift-logging"
+spec:
+  outputs:
+    - name: gcp-1
+      type: googleCloudLogging
+      secret:
+        name: gcp-secret
```
Comment (IshwarKanse) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:36]: For the gcp-secret we can give an example of creating the secret using the Google service account key.

`oc -n openshift-logging create secret generic gcp-secret --from-file google-application-credentials.json=SERVICE_ACCOUNT_KEY_FILE.json`

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      googleCloudLogging: <1>
+        projectId : "openshift-gce-devel"
+        logId : "app-gcp" <2>
+  pipelines:
+    - name: test-app
+      inputRefs:
```
Comment (IshwarKanse) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:42]: We can add the <3> point here for inputRefs with description. _Specify which log types to forward by using the pipeline: application, infrastructure, or audit._ Also remove the #- infrastructure

File: modules/cluster-logging-collector-log-forward-gcp.adoc (hunk 11 above)
Comment (anpingli) [reply modules/cluster-logging-collector-log-forward-gcp.adoc:36]: @libander  Could you add this step?

**Comment by libander** [comment]
Updated Preview: https://drive.google.com/file/d/1rZByPAvvYxnwdNZryCVoZ7bHRB3zkF5p/view?usp=sharing

**Review by anpingli — COMMENTED** [review]
/lgtm

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+.Prerequisites
+* {logging-title-uc} Operator 5.5.1 and later
+
+.Procedure
+
+. Create a secret using your link:https://cloud.google.com/iam/docs/creating-managing-service-account-keys[Google service account key].
++
+
+[subs="+quotes"]
```
Comment (bburt-rh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:24]: This needs a source tag `source,terminal` added before the `subs="+quotes"` per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+----
+oc -n openshift-logging create secret generic gcp-secret --from-file google-application-credentials.json=_<your_service_account_key_file.json>_
```
Comment (bburt-rh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:26]: The command needs an appropriate prompt ('$' or '#'),  per
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/cluster-logging-eventrouter-deploy.adoc [hunk 15]
```diff
@@ -124,8 +124,8 @@ parameters:
 <4> Creates a config map in the `openshift-logging` project to generate the required `config.json` file.
 <5> Creates a deployment in the `openshift-logging` project to generate and configure the Event Router pod.
 <6> Specifies the image, identified by a tag such as `v0.3`.
```
Comment (anpingli) [inline modules/cluster-logging-eventrouter-deploy.adoc:126]: @libander  Specifies the image, identified by a tag such as `v0.4`.

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+  name: "instance"
+  namespace: "openshift-logging"
+spec:
+  outputs:
+    - name: gcp-1
+      type: googleCloudLogging
+      secret:
+        name: gcp-secret
+      googleCloudLogging: <1>
+        projectId : "openshift-gce-devel"
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:44]:
```suggestion
      googleCloudLogging:
        projectId : "openshift-gce-devel" <1>
```
Would it be clearer to put callout #1 here instead?

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        logId : "app-gcp" <2>
+  pipelines:
+    - name: test-app
+      inputRefs: <3>
+        - application
+      outputRefs:
+        - gcp-1
+----
+<1> Set either a `projectId`, `folderId`, `organizationId`, or `billingAccountId` stanza, depending on where you want to store your logs in the link:https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy[GCP resource hierarchy].
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:53]:
```suggestion
<1> Set either a `projectId`, `folderId`, `organizationId`, or `billingAccountId` name-value pair, depending on where you want to store your logs in the link:https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy[GCP resource hierarchy].
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+<2> Set the `logId` value to add to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry].
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:54]:
```suggestion
<2> Set the value to add to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry].
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+.Procedure
+
+. Create a secret using your link:https://cloud.google.com/iam/docs/creating-managing-service-account-keys[Google service account key].
++
+[source,terminal,subs="+quotes"]
+----
+$ oc -n openshift-logging create secret generic gcp-secret --from-file google-application-credentials.json=_<your_service_account_key_file.json>_
+----
++
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:27]:
```suggestion

```

File: modules/cluster-logging-collector-log-forward-gcp.adoc [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+        - application
+      outputRefs:
+        - gcp-1
+----
+<1> Set either a `projectId`, `folderId`, `organizationId`, or `billingAccountId` stanza, depending on where you want to store your logs in the link:https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy[GCP resource hierarchy].
+<2> Set the `logId` value to add to the `logName` field of the link:https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry[Log Entry].
+<3> Specify which log types to forward by using the pipeline: `application`, `infrastructure`, or `audit`.
+
+[role="_additional-resources"]
+.Additional Resources
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-gcp.adoc:58]:
```suggestion
.Additional resources
```

File: modules/cluster-logging-collector-log-forward-gcp.adoc (hunk 18 above)
Comment (rolfedh) [reply modules/cluster-logging-collector-log-forward-gcp.adoc:54]: Annotations generally avoid repeating the name of the field they are about.

**Review by rolfedh — APPROVED** [review]

**Pull request merged**